src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -77,6 +77,8 @@ private enum ParameterSyntaxKind : byte
 
         internal SyntaxTree SyntaxTree => _syntaxRef == null ? null : _syntaxRef.SyntaxTree;
 
+        public sealed override bool IsDiscard => false;
+
```

Comment:
Are we making an assumption that this type is never used to represent lambda parameter? Consider adding an assert to the constructor that ```owner``` is not a LambdaSymbol with a comment that we rely on the assumption in ```IDiscard``` implementation.